docs/linear_system_solvers.rst

```diff
@@ -0,0 +1,61 @@
+Linear System Solvers
+=====================
+
+This section is concerned with solving problems of the form
+
+.. math::
+
+    Ax = b
+
+with unknown :math:`x` for a linear operator :math:`A` and vector :math:`b`.
+
+
+Iterative Refinement
+--------------------
+
+.. autosummary::
+  :toctree: _autosummary
+
+    jaxopt.IterativeRefinement
+
+
+Instantiating and running the solver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The class :class:`IterativeRefinement` aims to improve accuracy of linear solvers 
+from ``jax.scipy.sparse.linalg``.
+
+This is useful on poorly conditioned systems::
+
+    import jax
+    import jax.numpy as jnp
+    import numpy as onp
+    from jaxopt import IterativeRefinement
+
+    # ill-conditioned linear system
+    A = jnp.array([[3.9, 1.65], [6.845, 2.9]])
+    b = jnp.array([5.5, 9.7])
+    print(f"Condition number: {onp.linalg.cond(A):.0f}")
+    # Condition number: 4647
+
+    ridge = 1e-2
+    tol = 1e-7
+
+    x = jax.scipy.sparse.linalg.gmres(lambda x: jnp.dot(A, x), b, tol=tol)[0]
+    print(f"GMRES only error: {jnp.linalg.norm(A @ x - b):.7f}")
+    # GMRES only error: nan
+
+    def gmres_ridge(matvec_A, b):
+      def matvec(x):
+        return matvec_A(x) + ridge * x
+    return jax.scipy.sparse.linalg.gmres(matvec, b, tol=tol)[0]
+
+    x_ridge = gmres_ridge(lambda x: jnp.dot(A, x) + ridge * x, b)
+    print(f"GMRES+ridge error: {jnp.linalg.norm(A @ x_ridge - b):.7f}")
+    # GMRES+ridge error: 0.0500162
+
+    solver = IterativeRefinement(solve=gmres_ridge,
+                                 tol=tol, maxiter=100)
+    x_refined, state = solver.run(init_params=None, params_A=A, b=b)
```

Comment:
`params_A=A` is a bit confusing, maybe we should rename `params_A` to `A`, although I know `A` is not ideal either in the matvec case...